mapviz/src/mapviz.cpp

```diff
@@ -921,9 +921,27 @@ void Mapviz::SaveConfig()
 
 void Mapviz::ClearHistory()
 {
+  
+  ROS_DEBUG("Mapviz::ClearHistory()");
   for (auto& plugin: plugins_)
   {
-    plugin.second->ClearHistory();
+    if (plugin.second)
+    {
+      plugin.second->ClearHistory();
+      
+
+    }
+    else
+    {
+      // May not be necessary now, but good to have as a failsafe / debugging tool.  
+      // Attempting to clear empty map elements previously led to segfault
+      
+      // trying to acces which plugins crash also result in a segfault.
+      //QString qs = plugin.first->text(); 
+      //std::string s = qs.toUtf8().constData();
+      ROS_INFO("Clear Failed.");
+    }
+    
```

Comment:
Let's go ahead and remove the excess whitespace here, too.
```suggestion
  ROS_DEBUG("Mapviz::ClearHistory()");
  for (auto& plugin: plugins_)
  {
    plugin.second->ClearHistory();
    if (plugin.second)
    {
      plugin.second->ClearHistory();
    }
    else
    {
      // May not be necessary now, but good to have as a failsafe / debugging tool.  
      // Attempting to clear empty map elements previously led to segfault
      // trying to access which plugins crash also result in a segfault.
      ROS_INFO("Clear Failed.");
    }
```